apm-agent-plugins/apm-quartz-job-plugin/src/test/java/co/elastic/apm/agent/quartz/job/JobTransactionNameInstrumentationTest.java

```diff
@@ -0,0 +1,194 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.quartz.job;
+
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.BeforeClass;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.quartz.Job;
+import org.quartz.JobBuilder;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.SchedulerFactory;
+import org.quartz.SimpleScheduleBuilder;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.quartz.impl.JobDetailImpl;
+import org.quartz.impl.JobExecutionContextImpl;
+import org.quartz.impl.StdSchedulerFactory;
+
+import co.elastic.apm.agent.MockReporter;
+import co.elastic.apm.agent.bci.ElasticApmAgent;
+import co.elastic.apm.agent.configuration.SpyConfiguration;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.ElasticApmTracerBuilder;
+import co.elastic.apm.agent.quartz.job.JobTransactionNameInstrumentation;
+import co.elastic.apm.agent.report.ApmServerReporter;
+import net.bytebuddy.agent.ByteBuddyAgent;
+import org.springframework.scheduling.quartz.QuartzJobBean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+class JobTransactionNameInstrumentationTest {
+
+    private static MockReporter reporter;
+    private static ElasticApmTracer tracer;
+    private Scheduler scheduler = null;
+
+    @BeforeAll
+    static void setUpAll() {
+        reporter = new MockReporter();
+        tracer = new ElasticApmTracerBuilder()
+                .configurationRegistry(SpyConfiguration.createSpyConfig())
+                .reporter(reporter)
+                .build();
+        ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install(),
+                Collections.singletonList(new JobTransactionNameInstrumentation(tracer)));
+    }
+    void assertTests(JobDetail job) {
+    	assertThat(reporter.getTransactions().size()).isEqualTo(1);
+        assertThat(reporter.getTransactions().get(0).getName())
+        	.isEqualToIgnoringCase(String.format("%s.%s", job.getKey().getGroup(), job.getKey().getName()));
+    }
+    @Test
+    void testJobWithGroup() throws SchedulerException, InterruptedException {
+        JobDetail job = JobBuilder.newJob(TestJob.class)
+    			.withIdentity("dummyJobName", "group1").build();
+        Trigger trigger = TriggerBuilder
+        		.newTrigger()
+        		.withIdentity("myTrigger")
+        		.withSchedule(
+        				SimpleScheduleBuilder.repeatSecondlyForTotalCount(1, 1))
```

Comment:
There is an option to use `SimpleScheduleBuilder.simpleSchedule().withIntervalInMilliseconds()`, which allows you test multiple executions quickly, and `SimpleScheduleBuilder#withMisfireHandlingInstructionFireNow` for a single one (?)

Reply:
As the first execution fires instantly we do not need to change the code to millisecond interval. This only affects the interval between 2 executions.
withMisfireHandlingInstructionFireNow only affects the execution if the job was not executed(scheduler paused,...) so as soon the scheduler is available again the job will be executed instantly.